pkg/app/provider/wopi/wopi.go

```diff
@@ -142,16 +142,29 @@ func (p *wopiProvider) GetAppURL(ctx context.Context, resource *provider.Resourc
 	}
 
 	q.Add("appname", p.conf.AppName)
-	q.Add("appurl", p.appURLs["edit"][ext])
 
-	if p.conf.AppIntURL != "" {
-		q.Add("appinturl", p.conf.AppIntURL)
+	var viewAppURL string
+	if viewAppURLs, ok := p.appURLs["view"]; ok {
+		if viewAppURL, ok = viewAppURLs[ext]; ok {
+			q.Add("appviewurl", viewAppURL)
+		}
 	}
-	if viewExts, ok := p.appURLs["view"]; ok {
-		if url, ok := viewExts[ext]; ok {
-			q.Add("appviewurl", url)
+	if editAppURLs, ok := p.appURLs["edit"]; !ok {
+		if editAppURL, ok := editAppURLs[ext]; ok {
```

Comment:
If this `ok` is false, we won't add the `appurl` query

Reply:
It will now raise an error if neither "appurl" nor "appviewurl" is set. And it will set the "appurl" header to the same value as "appviewurl" if it is empty.